### PR TITLE
Use Selenium to test Chrome

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses https://github.com/aspnet/KestrelHttpServer/issues/3075
I still want to evaluate whether this version is more reliable. It ended up being a test only change so I was hoping to resolve the flakiness in 2.2 as well.

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
I've ran the converted Selenium tests 50 times without seeing any failures. Previously I was seeing a test failure with headless chrome 1 in 10-20 runs. It also turns out Selenium tests run faster in 8.6 seconds vs 11.4 seconds before the change.

File: test/Kestrel.Transport.Sockets.FunctionalTests/Kestrel.Transport.Sockets.FunctionalTests.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <DefineConstants Condition="$([MSBuild]::IsOSPlatform('OSX'))">$(DefineConstants);MACOS</DefineConstants>
     <ServerGarbageCollection>true</ServerGarbageCollection>
     <TestGroupName>Sockets.FunctionalTests</TestGroupName>
+    <NoWarn>CS8002</NoWarn>
```
Comment (JunTaoLuo) [inline test/Kestrel.Transport.Sockets.FunctionalTests/Kestrel.Transport.Sockets.FunctionalTests.csproj]: Unfortunately Selenium.WebDriver isn't strong named so I have to ignore this csc error. Feels strange that we don't do this anywhere else though.

File: build/dependencies.props [hunk 2]
```diff
@@ -36,6 +36,8 @@
     <MoqPackageVersion>4.10.0</MoqPackageVersion>
     <NETStandardLibrary20PackageVersion>2.0.3</NETStandardLibrary20PackageVersion>
     <NewtonsoftJsonPackageVersion>11.0.2</NewtonsoftJsonPackageVersion>
+    <SeleniumWebDriverChromeDriverPackageVersion>2.43.0</SeleniumWebDriverChromeDriverPackageVersion>
```
Comment (JunTaoLuo) [inline build/dependencies.props]: cc @Eilon, these are new dependencies on third party software. Can we get these through OSS?

File: test/Kestrel.Transport.Sockets.FunctionalTests/Kestrel.Transport.Sockets.FunctionalTests.csproj [hunk 3]
```diff
@@ -7,6 +7,7 @@
     <DefineConstants Condition="$([MSBuild]::IsOSPlatform('OSX'))">$(DefineConstants);MACOS</DefineConstants>
     <ServerGarbageCollection>true</ServerGarbageCollection>
     <TestGroupName>Sockets.FunctionalTests</TestGroupName>
+    <WarningsNotAsErrors>CS8002;$(WarningsNotAsErrors)</WarningsNotAsErrors>
```
Comment (halter73) [inline test/Kestrel.Transport.Sockets.FunctionalTests/Kestrel.Transport.Sockets.FunctionalTests.csproj]: Does this mean one or both of the Selenium packages aren't strong named?

Comment (JunTaoLuo) [reply test/Kestrel.Transport.Sockets.FunctionalTests/Kestrel.Transport.Sockets.FunctionalTests.csproj]: Just Selenium.WebDriver

File: build/dependencies.props (hunk 2 above)
Comment (Eilon) [reply build/dependencies.props]: Good to go.

**Comment by JunTaoLuo** [comment]
ping @halter73 @Tratcher

File: test/Kestrel.Transport.FunctionalTests/ChromeTests.cs [hunk 4]
```diff
@@ -40,33 +42,33 @@ public class ChromeTests : LoggedTest
...
-                $"--virtual-time-budget=10000 " +
-                $"--log-net-log={NetLogPath} " +
-                $"--trace-startup --trace-startup-file={StartupLogPath} " +
-                $"--trace-shutdown --trace-shutdown-file={ShutdownLogPath}";
+            ChromeArgs = new [] {
+                $"--headless ",
+                $"--no-sandbox ",
+                $"--disable-gpu ",
+                $"--allow-insecure-localhost ",
+                $"--ignore-certificate-errors --enable-features=NetworkService ",
```
Comment (halter73) [inline test/Kestrel.Transport.FunctionalTests/ChromeTests.cs]: Why are you including multiple argument flags in a single array element?

File: test/Kestrel.Transport.FunctionalTests/ChromeTests.cs [hunk 5]
```diff
@@ -7,6 +7,7 @@
 using System.Diagnostics;
```
Comment (halter73) [inline test/Kestrel.Transport.FunctionalTests/ChromeTests.cs]: You might want to move this to Kestrel.Transport.Sockets.FunctionalTests so it doesn't get run for the libuv transport functional tests now that #3091 has been merged.

File: test/Kestrel.Transport.FunctionalTests/ChromeTests.cs (hunk 4 above)
Comment (JunTaoLuo) [reply test/Kestrel.Transport.FunctionalTests/ChromeTests.cs]: Good point, I didn't notice when I was updating the format.

**Comment by JunTaoLuo** [comment]
cc @muratg this is the change we'd like to make in 2.2 to make headless chrome tests more stable.

File: build/dependencies.props (hunk 2 above)
Comment (JunTaoLuo) [reply build/dependencies.props]: This can now be down non-atomically, apparently.

**Pull request merged**